data.c

```diff
@@ -369,6 +358,7 @@ backup_data_file(const char *from_root,
 		int		upper_length;
 
 		header.block = blknum;
+		header.endpoint = false;
```

Comment:
I think endpoint is not lastpage

Reply:
Hmm, yeah.  The page before the dummy one is actually the last page of the relation, so perhaps this should be called `endpoint` after all.

Reply:
Thanks for reviewing!

Sure, but I'm a little bit worried about this flag is true at the "lastpage + 1" in the context of PostgreSQL page. (Because this patch adds extra page for the flag.)

Reply:
I'll close this comment in favor of leaving `endpoint` unchanged, that is, doing nothing about this suggestion.  `endpoint` is fine.